src/aircraft_gui.cpp

```diff
@@ -34,7 +34,7 @@ void DrawAircraftDetails(const Aircraft *v, const Rect &r)
 	int y = r.top;
 	for (const Aircraft *u = v; u != nullptr; u = u->Next()) {
 		if (u->IsNormalAircraft()) {
-			SetDParam(0, u->engine_type);
+			SetDParam(0, u->engine_type | (static_cast<uint64>(EngineNameContext::VehicleDetails) << 32));
```

Comment:
Might it be useful to hide the exact details of the bitstuffing?
```
uint64 EngineNameContextPack(EngineNameContext context, EngineID engine_id) {
    return (static_cast<uint64>(context) << 32) | engine_id;
}

std::tuple<EngineNameContext, EngineID> EngineNameContextUnpack(uint64 packed) {
    return { static_cast<EngineNameContext>(packed >> 32), static_cast<EngineID>(packed) };
}
```

Even though I whine a lot about Java missing really useful C++ features at work, here C++ kind of misses a feature of Java. The ability to add functions to enumerations. It would have been really nice to be able to write `EngineNameContext::VehicleDetails.Pack(u->engine_type)` or something similar. With a custom one for the purchase list's indent that allows two parameters.

Reply:
Went for a pack function, but no explicit unpack as it is done only in once place.